### PR TITLE
test: Condition tests for caching-materials-manager-browser

### PR DESCRIPTION
**Pull request by seebees**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

# Check any applicable:
- [ ] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

File: modules/caching-materials-manager-browser/test/caching_materials_manager_browser.test.ts [hunk 1]
```diff
@@ -59,4 +59,28 @@ describe('WebCryptoCachingMaterialsManager', () => {
     expect(test._maxBytesEncrypted).to.equal(maxBytesEncrypted)
     expect(test._maxMessagesEncrypted).to.equal(maxMessagesEncrypted)
   })
+
+  it('Precondition: A partition value must exist for WebCryptoCachingMaterialsManager.', () => {
```
Comment (lavaleri) [inline modules/caching-materials-manager-browser/test/caching_materials_manager_browser.test.ts]: Does this test also need to be added to caching_materials_manager_node.test.ts ?

File: modules/caching-materials-manager-browser/test/caching_materials_manager_browser.test.ts [hunk 2]
```diff
@@ -59,4 +59,28 @@ describe('WebCryptoCachingMaterialsManager', () => {
...
+    const maxAge = 10
+    const test = new WebCryptoCachingMaterialsManager({
+      backingMaterials: keyring,
+      cache,
+      maxAge
+    })
+    /* Binary data is being transformed to utf-8.
+     * This means it is highly likely to be less than 64 characters long.
+     */
+    expect(test._partition).to.be.a('string').and.to.have.length.greaterThan(50)
```
Comment (lavaleri) [inline modules/caching-materials-manager-browser/test/caching_materials_manager_browser.test.ts]: What is the likelihood that 64 bytes of random data -> UTF-8 will be less than 50 characters? Enough that we will "never" see this test fail?

File: modules/caching-materials-manager-browser/test/caching_materials_manager_browser.test.ts (hunk 1 above)
Comment (seebees) [reply modules/caching-materials-manager-browser/test/caching_materials_manager_browser.test.ts]: Yes, and it is in a different PR

File: modules/caching-materials-manager-browser/test/caching_materials_manager_browser.test.ts (hunk 2 above)
Comment (seebees) [reply modules/caching-materials-manager-browser/test/caching_materials_manager_browser.test.ts]: Sigh... I "fixed" this in https://github.com/awslabs/aws-encryption-sdk-javascript/pull/178/files#diff-2e3dd129c30c2285d72b61362efd41b9R58

Once that is merged.  I'll update the test... utf8 on random data is.... sub optimal.

File: modules/caching-materials-manager-browser/test/caching_materials_manager_browser.test.ts [hunk 3]
```diff
@@ -59,4 +59,28 @@ describe('WebCryptoCachingMaterialsManager', () => {
...
+    const keyring = new TestKeyring()
+    const cache = 'cache' as any
+    const maxAge = 10
+    const test = new WebCryptoCachingMaterialsManager({
+      backingMaterials: keyring,
+      cache,
+      maxAge
+    })
+    /* Binary data is being transformed to utf-8.
+     * This means it is highly likely to be less than 64 characters long.
```
Comment (lavaleri) [inline modules/caching-materials-manager-browser/test/caching_materials_manager_browser.test.ts]: This comment is no longer needed

**Pull request merged**